### ISSUE TITLE
Cython not declared as requirement

### ISSUE THREAD

**Issue by jbarnoud** [MDAnalysis/mdanalysis#768]
Since #692, cython is required to install MDAnalysis from the git repository. Yet, cython is not listed as a requirement.

Cython should be listed as a requirement if `setup.py` is configure to delete the cython files.
### Expected behaviour

The dev version of MDAnalysis get install with `pip install <path-to-MDAnalysis>/package`.
### Actual behaviour

```
Obtaining file:///home/jon/dev/mda3/package
    Complete output from command python setup.py egg_info:
    Attempting to autodetect OpenMP support... Compiler supports OpenMP
    Will attempt to use Cython.
    Couldn't find a Cython installation. Not recompiling cython extensions.
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/home/jon/dev/mda3/package/setup.py", line 372, in <module>
        exts, cythonfiles = extensions(config)
      File "/home/jon/dev/mda3/package/setup.py", line 351, in extensions
        "failed/disabled Cython build.".format(source))
    IOError: Source file 'MDAnalysis/coordinates/dcdtimeseries.c' not found. This might be caused by a missing Cython install, or a failed/disabled Cython build.
```
### Code to reproduce the behaviour

``` python
pip install <path-to-MDAnalysis>/package`
```
### Currently version of MDAnalysis:

0.14.1.dev0

**Comment by orbeckst** [comment]
Good point, just encountered this problem (see dev list)  but did not realize that I should have filed an issue. Thanks!

**Comment by kain88-de** [comment]
Don't we need to handle this the same as numpy? We need to have cython available very early in the build process for configurations.

**Comment by jbarnoud** [comment]
@kain88-de Yes, I think we do. Everything related to Cython should be delayed until after the "setup_requires" step is done.

**Comment by orbeckst** [comment]
Maybe chat to @mnmelo on how to do the delayed checking?

--
Oliver Beckstein
email: orbeckst@gmail.com


Am Mar 17, 2016 um 19:57 schrieb Jonathan Barnoud <notifications@github.com>:

**Comment by mnmelo** [comment]
I'm on it. Seems to be possible to handle the same way. Maybe I can even further generalize the lazy dependency model.

**Comment by jbarnoud** [comment]
@mnmelo Glad to leave that to you.

**Comment by kain88-de** [comment]
Solved in #1134
